fineract-provider/src/main/java/org/apache/fineract/infrastructure/codes/service/CodeValueReadPlatformServiceImpl.java

```diff
@@ -54,12 +58,17 @@ public String schema() {
         public CodeValueData mapRow(final ResultSet rs, @SuppressWarnings("unused") final int rowNum) throws SQLException {
 
             final Long id = rs.getLong("id");
-            final String value = rs.getString("value");
+            final Long codeId = rs.getLong("codeId");
+            String value = rs.getString("value");
             final Integer position = rs.getInt("position");
             final String description = rs.getString("description");
             final boolean isActive = rs.getBoolean("isActive");
             final boolean mandatory = rs.getBoolean("mandatory");
 
+            if (codeId.equals(CodevalueJSONinputParams.COUNTRYCODE.getValueLong())) {
```

Comment:
I'm not convinced that architecturally the translation should be here.

Let's think about data layer and presentation layer. Where the business logic is implemented, or the database is touched, let's call that the data layer. On that level, we should definitely not take care of translation.

When we are presenting the data - i.e. returning the value through an API - we definitely could translate the country code into a readable country name. However, even in that case, I simply wouldn't just stick with the readable country name on the API response but include the country code as well so if the users of the API would like to differentiate between countries, they don't need to hardcode the readable country names in different languages.